### PR TITLE
Docs: FAQ

### PR DESCRIPTION
**Pull request by ax3l**
Start a usage and a developer FAQ section.

Thanks a ton to @RTSandberg for the great questions!

### CODE REVIEW

**Review by RTSandberg — APPROVED** [review]
These look good to me!  Hopefully other developers find this helpful.

**Comment by ax3l** [comment]
I bet they will, thank you so much!

File: Docs/source/developers/faq.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+-------------------
+
+It's a C++ `floating-point literal <https://en.cppreference.com/w/cpp/language/floating_literal>`__ for zero of type ``amrex::Real``.
+
+We use `literals <https://en.cppreference.com/w/cpp/language/expressions#Literals>`__ to define constants with a specific type, in that case the zero-value.
+There is also ``0.0_prt``, which is a literal zero of type ``amrex::ParticleReal``.
+In std C++, you know: ``0.0`` (literal ``double``), ``0.0f`` (literal ``float``) and ``0.0L`` (literal ``long double``).
+We do not use use those, so that we can configure floating point precision at compile time and use different precision for fields (``amrex::Real``) and particles (``amrex::ParticleReal``).
+
+You can also things like ``42.0_prt`` if you like to have another value than zero.
```
Comment (EZoni) [inline Docs/source/developers/faq.rst]: Maybe a word is missing here: "code", "use"?
```suggestion
You can also code things like ``42.0_prt`` if you like to have another value than zero.
```

**Review by EZoni — COMMENTED** [review]
Thank you, Axel! This is a great starting point. I just left one small comment.

File: Docs/source/developers/faq.rst (hunk 1 above)
Comment (ax3l) [reply Docs/source/developers/faq.rst]: Oh yes, this sentence no verb
```suggestion
You can also write things like ``42.0_prt`` if you like to have another value than zero.
```

**Pull request merged**